stdlib/int16.mli

```diff
@@ -0,0 +1,171 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                         The OCaml programmers                          *)
+(*                                                                        *)
+(*   Copyright 2025 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** 16-bit integers.
+
+    This module provides operations on the type [t] of signed 16-bit integers.
+    All arithmetic operations over [t] are taken modulo 2{^16}.
+
+    Internally, values of type [t] are represented as [int].
+
+    @since 5.4 *)
+
+type t
```

Comment:
I think we should have either

```ocaml
type t = private int
```

or 

```ocaml
type t [@@immediate]
```

here, and similarly for `int8.mli`